modules/tinymce/src/core/main/ts/EditorSettings.ts

```diff
@@ -73,73 +63,43 @@ const getSectionConfig = (sectionResult: SectionResult, name: string) => {
   return hasSection(sectionResult, name) ? sectionResult.sections()[name] : {};
 };
 
-const getToolbarMode = (settings: RawEditorSettings, defaultVal: ToolbarMode) =>
+const getToolbarMode = (options: RawEditorOptions, defaultVal: ToolbarMode) =>
   // If toolbar_mode is unset by the user, fall back to:
-  Obj.get(settings, 'toolbar_mode').getOr(defaultVal);
-
-const getDefaultSettings = (settings: RawEditorSettings, id: string, documentBaseUrl: string, isTouch: boolean, editor: Editor): RawEditorSettings => {
-  const baseDefaults: RawEditorSettings = {
-    id,
-    theme: 'silver',
-    toolbar_mode: getToolbarMode(settings, 'floating'),
-    plugins: '',
-    document_base_url: documentBaseUrl,
-    add_form_submit_trigger: true,
-    submit_patch: true,
-    add_unload_trigger: true,
-    convert_urls: true,
-    relative_urls: true,
-    remove_script_host: true,
-    object_resizing: true,
-    doctype: '<!DOCTYPE html>',
-    visual: true,
-
-    // See: http://www.w3.org/TR/CSS2/fonts.html#propdef-font-size
-    font_size_legacy_values: 'xx-small,small,medium,large,x-large,xx-large,300%',
-    forced_root_block: 'p',
-    hidden_input: true,
-    inline_styles: true,
-    convert_fonts_to_spans: true,
-    indent: true,
-    indent_before: 'p,h1,h2,h3,h4,h5,h6,blockquote,div,title,style,pre,script,td,th,ul,ol,li,dl,dt,dd,area,table,thead,' +
-    'tfoot,tbody,tr,section,summary,article,hgroup,aside,figure,figcaption,option,optgroup,datalist',
-    indent_after: 'p,h1,h2,h3,h4,h5,h6,blockquote,div,title,style,pre,script,td,th,ul,ol,li,dl,dt,dd,area,table,thead,' +
-    'tfoot,tbody,tr,section,summary,article,hgroup,aside,figure,figcaption,option,optgroup,datalist',
-    entity_encoding: 'named',
-    // Note: Don't bind here, as the binding is handled via the `url_converter_scope`
-    // eslint-disable-next-line @typescript-eslint/unbound-method
-    url_converter: editor.convertURL,
-    url_converter_scope: editor
+  Obj.get(options, 'toolbar_mode').getOr(defaultVal);
+
+const getDefaultOptions = (options: RawEditorOptions, isTouch: boolean): RawEditorOptions => {
+  const baseDefaults: RawEditorOptions = {
+    toolbar_mode: getToolbarMode(options, 'floating')
   };
 
   return {
     ...baseDefaults,
-    ...isTouch ? defaultTouchSettings : { }
+    ...isTouch ? defaultTouchOptions : { }
   };
 };
 
-const getDefaultMobileSettings = (mobileSettings: RawEditorSettings, isPhone: boolean): RawEditorSettings => {
-  const defaultMobileSettings: RawEditorSettings = {
+const getDefaultMobileOptions = (mobileOptions: RawEditorOptions, isPhone: boolean): RawEditorOptions => {
```

Comment:
This logic should be able to be removed once we migrate the theme/plugins as it will be responsible for setting the defaults when registering.